pkg/kp/consulutil/watch.go

```diff
@@ -55,6 +55,48 @@ func WatchPrefix(
 	}
 }
 
+// WatchSingle has the same semantics as WatchPrefix, but for a single key in
+// Consul.
```

Comment:
This method won't necessarily signal that a key has been deleted if it was quickly recreated afterwards.  For the application of this routine, that doesn't matter, but could you document that as a known issue?